sdv/multi_table/base.py

```diff
@@ -82,3 +83,74 @@ def update_table_parameters(self, table_name, table_parameters):
     def get_metadata(self):
         """Return the ``MultiTableMetadata`` for this synthesizer."""
         return self.metadata
+
+    def _validate_foreign_keys(self, data):
+        errors = []
+        for relation in self.metadata._relationships:
+            child_table = data[relation['child_table_name']]
+            parent_table = data[relation['parent_table_name']]
+            child_row = child_table[relation['child_foreign_key']]
+            parent_row = parent_table[relation['parent_primary_key']]
+            missing_values = child_row[~child_row.isin(parent_row)].unique()
+            if any(missing_values):
+                message = ', '.join(missing_values[:5].astype(str))
+                if len(missing_values) > 5:
+                    message = f'({message}, + more).'
+                else:
+                    message = f'({message}).'
+
+                errors.append(
+                    f"Error: foreign key column '{relation['child_foreign_key']}' contains "
+                    f'unknown references: {message}'
+                )
+
+        return errors
+
+    def validate(self, data):
+        """Validate data.
+
+        Args:
+            data (dict):
+                A dictionary with key as table name and ``pandas.DataFrame`` as value to validate.
+
+        Raises:
+            ValueError:
+                Raised when data is not of type pd.DataFrame.
+            InvalidDataError:
+                Raised if:
+                    * foreign key does not belong to a primay key
+                    * data columns don't match metadata
+                    * keys have missing values
+                    * primary or alternate keys are not unique
+                    * context columns vary for a sequence key
+                    * values of a column don't satisfy their sdtype
+        """
+        errors = []
+        missing_tables = set(self.metadata._tables) - set(data)
+        if missing_tables:
+            if len(missing_tables) > 1:
+                raise InvalidDataError([
+                    f'The provided data is missing the tables {missing_tables}.'
+                ])
+            else:
```

Comment:
I don't think we need to be this careful with error messages (adding another error message just for the case where it's plural). I at least always use plural for a set of items. @amontanez24 any opinions on it?

Reply:
Yeah I think it's fine to just use plural and print it as a list or set